Microsoft.HealthVault.Fhir/Transformers/MedicationRequestToHealthVault.cs

```diff
@@ -0,0 +1,193 @@
+            {
+                var practitioner = ExtractEmbeddedPractitioner(medicationRequest);
+                var prescription = new Prescription(practitioner.ToHealthVault());
+                if (medicationRequest.AuthoredOnElement != null)
+                {
+                    var dt = medicationRequest.AuthoredOnElement.ToDateTimeOffset();
+                    prescription.DatePrescribed = new ApproximateDateTime(
+                            new ApproximateDate(dt.Year, dt.Month, dt.Day),
+                            new ApproximateTime(dt.Hour, dt.Minute, dt.Second, dt.Millisecond));
+                }
+
+                MedicationRequest.DispenseRequestComponent dispenseRequest = medicationRequest.DispenseRequest;
+                if (dispenseRequest != null)
+                {
+                    var numerator = dispenseRequest.Quantity;
+
+                    if (numerator != null)
+                    {
+                        var structuredMeasurement = new StructuredMeasurement
+                        {
+                            Value = (double)numerator.Value,
+                            Units = CodeToHealthVaultHelper.CreateCodableValueFromQuantityValues(numerator.System,
+                                   numerator.Code, numerator.Unit)
+                        };
+                        prescription.AmountPrescribed = new GeneralMeasurement();
+                        prescription.AmountPrescribed.Structured.Add(structuredMeasurement);
+                    }
+
+                    if (dispenseRequest.NumberOfRepeatsAllowed.HasValue)
+                    {
+                        prescription.Refills = dispenseRequest.NumberOfRepeatsAllowed;
+                    }
+                    else
+                    {
+                        prescription.Refills = 0;
+                    }
+
+                    prescription.DaysSupply = GetDaysFromDuration(dispenseRequest.ExpectedSupplyDuration);
+
+                    FhirDateTime end = dispenseRequest.ValidityPeriod?.EndElement;
+                    if (end != null)
+                    {
+                        var endDate = end.ToDateTimeOffset();
+                        prescription.PrescriptionExpiration = new HealthServiceDate(endDate.Year
+                            , endDate.Month, endDate.Day);
+                    }
+                }
+                if (medicationRequest.DosageInstruction.Any())
+                {
+                    var hasInstruction = medicationRequest.DosageInstruction.Any(
+                        dosage => dosage.AdditionalInstruction.Any());
+                    if (hasInstruction)
+                    {
+                        var instruction = medicationRequest.DosageInstruction.First(
+                         dosage => dosage.AdditionalInstruction.Any()).AdditionalInstruction.First();
+                        prescription.Instructions = instruction.ToCodableValue();//.GetCodableValue();
+                    }
```

Comment:
Consider refactoring this to avoid using First so you can avoid exceptions happening when items are not found. Something like this might work better:

     var dosageInstruction = medicationRequest.DosageInstruction.FirstOrDefault(
                        dosage => dosage.AdditionalInstruction.Any());
                    if (dosageInstruction != null)
                    {
                        var instruction = dosageInstruction.AdditionalInstruction.FirstOrDefault();
                        prescription.Instructions = instruction?.ToCodableValue();
                    }

Also, remove the comment in line 90, //GetCodableValue() if it is not longer needed.

Reply:
It won't throw any exceptions.
But your code looks cleaner, I shall change this.